File: src/main/java/cz/vhromada/catalog/web/mapper/ProgramMapper.java
package cz.vhromada.catalog.web.mapper;

import cz.vhromada.catalog.entity.Program;
import cz.vhromada.catalog.web.fo.ProgramFO;

import org.mapstruct.Mapper;

/**
 * An interface represents mapper for programs.
 *
 * @author Vladimir Hromada
 */
@Mapper
public interface ProgramMapper {

    /**
     * Returns FO for program.
     *
     * @param source program
     * @return FO for program
     */
    ProgramFO map(Program source);

    /**
     * Returns program.
     *
     * @param source FO for program
     * @return program
     */
    Program mapBack(ProgramFO source);

}

File: src/main/kotlin/cz/vhromada/catalog/web/mapper/ProgramMapper.kt
package cz.vhromada.catalog.web.mapper

import cz.vhromada.catalog.entity.Program
import cz.vhromada.catalog.web.fo.ProgramFO

/**
 * An interface represents mapper for programs.
 *
 * @author Vladimir Hromada
 */
interface ProgramMapper {

    /**
     * Returns FO for program.
     *
     * @param source program
     * @return FO for program
     */
    fun map(source: Program): ProgramFO

    /**
     * Returns program.
     *
     * @param source FO for program
     * @return program
     */
    fun mapBack(source: ProgramFO): Program

}
